spec/controllers/v2/application_controller_spec.rb

```diff
@@ -25,10 +24,80 @@
       end
     end
 
-    it 'returns true with correct permissions' do
-      subject.class.instance_variable_set(:@action_permissions, { index: Rbac::COMPLIANCE_VIEWER })
+    context 'with sufficient permissions' do
+      before do
+        permissions = [
+          RBACApiClient::Access.new(
+            permission: Rbac::INVENTORY_HOSTS_READ,
+            resource_definitions: []
+          ),
+          RBACApiClient::Access.new(
+            permission: Rbac::COMPLIANCE_VIEWER,
+            resource_definitions: []
+          )
+        ]
+        allow(Rbac).to receive(:load_user_permissions).and_return(permissions)
+      end
+
+      it 'returns true with correct permissions' do
+        subject.class.instance_variable_set(:@action_permissions, { index: Rbac::COMPLIANCE_VIEWER })
+
+        expect(subject.send(:rbac_allowed?)).to be_truthy
+      end
+    end
+
+    context 'with insufficient permission to inventory' do
+      before do
+        permissions = [
+          RBACApiClient::Access.new(
+            permission: 'inventory:groups:read', # insufficient to access hosts
+            resource_definitions: []
+          ),
+          RBACApiClient::Access.new(
+            permission: Rbac::COMPLIANCE_VIEWER,
+            resource_definitions: []
+          )
+        ]
+        allow(Rbac).to receive(:load_user_permissions).and_return(permissions)
+      end
+
+      it 'results in rejecting access because inventory:groups:read is not sufficient' do
+        subject.class.instance_variable_set(:@action_permissions, { index: Rbac::COMPLIANCE_VIEWER })
+
+        expect(subject.send(:rbac_allowed?)).to be_falsey
+      end
 
-      expect(subject.send(:rbac_allowed?)).to be_truthy
+      before do
+        permissions = [
+          RBACApiClient::Access.new(
+            permission: Rbac::INVENTORY_HOSTS_READ,
+            resource_definitions: []
+          ) # insufficient to access hosts
+        ]
+        allow(Rbac).to receive(:load_user_permissions).and_return(permissions)
+      end
+
+      it 'results in rejecting access because compliance:policy:read is missing' do
+        subject.class.instance_variable_set(:@action_permissions, { index: Rbac::COMPLIANCE_VIEWER })
+
+        expect(subject.send(:rbac_allowed?)).to be_falsey
+      end
```

Comment:
Before blocks aren't decorators, they are always applied to the whole describe/context block and not just the method afterwards.